File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    sum = 0
    for i in range(len(s)):
        if(sum <= 0):
            if(s[i] == ")"):
                return False
        if(s[i] == "("):
            sum += 1
        else:
            sum -= 1
    if(sum == 0):
        return True
    else:
        return False
    return answer

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        int left = 0;
        int right = 0;
        
        for(int i = 0 ; i < s.length() ; i++) {
            if(s.charAt(i) == ')') {
                if(left == 0) {
                    answer = false;
                }
                else {
                    left -= 1;
                }
            }
            else {
                left += 1;
            }
        }
        
        if(answer) {
            if(left == right) {
                answer = true;
            }
            else {
                answer = false;
            }   
        }

        return answer;
    }
}